src/context/AppContextProvider.tsx

```diff
@@ -0,0 +1,256 @@
+import {
+  FC, useState, useEffect, MouseEvent, useMemo,
+} from 'react';
+import {
+  deleteTodo, getTodos, patchTodo,
+} from '../api/todos';
+import { Todo, Filter } from '../types';
+import { AppContext } from './AppContext';
+import { ErrorMessage } from '../types/ErrorMessages';
+
+type Props = React.PropsWithChildren;
+
+export const AppContextProvider: FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState<Filter>(Filter.ALL);
+  const [showError, setShowError] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todosBeingLoaded, setTodosBeingLoaded] = useState<number[]>([]);
+  const [todoInEdit, setTodoInEdit] = useState<number | null>(null);
+
+  // CONSTS
+
+  // CALCULATING ACTIVE & COMPLETED TODOS
+  const activeTodosNum = todos.reduce((acc, curr) => {
```

Comment:
i think filter() and length will be cleaner
or even u can ommit length and just left completedTodos because u use it too in lane 133

Reply:
done ✅